parquet-thrift/src/main/java/org/apache/parquet/thrift/struct/CompatibilityChecker.java

```diff
@@ -142,6 +204,11 @@ private boolean firstIsMoreRestirctive(ThriftField.Requirement firstReq, ThriftF
   public void visit(ThriftType.StructType newStruct) {
     ThriftType.StructType currentOldType = ((ThriftType.StructType) oldType);
     short oldMaxId = 0;
+
+    if (newStruct.getChildren().size() == 0) {
+      report.emptyStruct("new struct is an empty struct: " + path);
```

Comment:
this might be a little misleading, maybe "encountered an empty struct" or something like that? users don't know that this variable is newStruct so new doesn't probably mean much to them